src/core/EdFi.DataManagementService.Core/Validation/DocumentValidator.cs

```diff
@@ -184,10 +184,36 @@ string[] segments
             return jsonObject;
         }
 
-        var node = jsonObject[segments[0]];
-        Trace.Assert(node != null, "PointerSegment not found on JsonObject");
-        var nodeObj = node.AsObject();
-        nodeObj.RemoveProperty(segments.Skip(1).ToArray());
+        var currentSegment = segments[0];
+        var remainingSegments = segments.Skip(1).ToArray();
+        var node = jsonObject[currentSegment];
+
+        Trace.Assert(node != null, $"PointerSegment '{currentSegment}' not found on JsonObject");
+
+        if (node is JsonObject nodeObj)
+        {
+            nodeObj.RemoveProperty(remainingSegments);
+        }
+        else if (node is JsonArray nodeArray && int.TryParse(remainingSegments[0], out int index))
```

Comment:
I didn't see any unit tests that cover this branch where it is an array. Can you add one?